src/ruby.rs

```diff
@@ -51,94 +51,319 @@ extern {
+
+    /// The Ruby [`RegexpError`](https://ruby-doc.org/core-2.5.1/RegexpError.html) exception class
     pub static rb_eRegexpError: VALUE;
+
+    /// The Ruby [`EncodingError`](https://ruby-doc.org/core-2.5.1/EncodingError.html) exception class
     pub static rb_eEncodingError: VALUE;
+
+    /// The Ruby [`Encoding::CompatibilityError`](https://ruby-doc.org/core-2.5.1/Encoding/CompatibilityError.html) exception class
     pub static rb_eEncCompatError: VALUE;
 
+    /// The Ruby [`ScriptError`](https://ruby-doc.org/core-2.5.1/ScriptError.html) exception class
     pub static rb_eScriptError: VALUE;
+
+    /// The Ruby [`NameError`](https://ruby-doc.org/core-2.5.1/NameError.html) exception class
     pub static rb_eNameError: VALUE;
+
+    /// The Ruby [`SyntaxError`](https://ruby-doc.org/core-2.5.1/SyntaxError.html) exception class
     pub static rb_eSyntaxError: VALUE;
+
+    /// The Ruby [`LoadError`](https://ruby-doc.org/core-2.5.1/LoadError.html) exception class
     pub static rb_eLoadError: VALUE;
 
+    /// The Ruby [`Math::DomainError`](https://ruby-doc.org/core-2.5.1/Math/DomainError.html) exception class
     pub static rb_eMathDomainError: VALUE;
 
+
+    /// Convert a null terminated ASCII encoded character pointer to an [`ID`](struct.id.html)
+    ///
+    /// # Safety
+    ///
+    /// `cstr` must be ASCII encoded
+    ///
+    /// # Defined In
+    ///
+    /// <https://github.com/ruby/ruby/blob/v2_5_1/include/ruby/ruby.h#L1728>
     pub fn rb_intern(cstr: *const c_char) -> ID;
+
+    /// Convert an ASCII encoded character pointer of the given length to an [`ID`](struct.id.html)
+    ///
+    /// [`rb_intern_const`](https://github.com/ruby/ruby/blob/v2_5_1/include/ruby/ruby.h#L1757-L1760)
+    /// is more or less an alias of this.
+    ///
+    /// # Safety
+    ///
+    /// `cstr` must be ASCII encoded
+    ///
+    /// # Defined In
+    ///
+    /// <https://github.com/ruby/ruby/blob/v2_5_1/include/ruby/ruby.h#L1729>
     pub fn rb_intern2(ptr: *const c_char, len: c_long) -> ID;
+
+    /// Convert a Ruby [`String`](https://ruby-doc.org/core-2.5.1/String.html) [`VALUE`](struct.VALUE.html)
+    /// to an [`ID`](struct.id.html)
+    ///
+    /// # Safety
+    ///
+    /// Unclear what happens if you pass a `VALUE` that is not a Ruby `String`.
+    ///
+    /// # Defined In
+    ///
+    /// <https://github.com/ruby/ruby/blob/v2_5_1/include/ruby/ruby.h#L1730>
     pub fn rb_intern_str(string: VALUE) -> ID;
 
+    /// Convert a Ruby [`Symbol`](https://ruby-doc.org/core-2.5.1/Symbol.html) [`VALUE`](struct.VALUE.html)
+    /// to an [`ID`](struct.id.html)
+    ///
+    /// [`RB_SYM2ID`](https://github.com/ruby/ruby/blob/v2_5_1/include/ruby/ruby.h#L381) and
+    /// [`SYM2ID`](https://github.com/ruby/ruby/blob/v2_5_1/include/ruby/ruby.h#L386) are currently
+    /// aliases for this.
+    ///
+    /// # Safety
+    ///
+    /// Raises a [`rb_eTypeError`](static.rb_eTypeError.html) if `symbol` is not a Ruby `Symbol`.
```

Comment:
* Raises the following exceptions:
  * ...

It _may_ be worth splitting exceptions into a sub-section even, with some standard boilerplate stating exceptions are highly unsafe without proper unwinding

Reply:
@sgrif how do you want to document this?